### ISSUE TITLE
OnFlushCompleted should update file discardable size

### ISSUE THREAD

**Issue by yiwu-arbug** [pingcap/titan#86]
Currently we only update file discardable size in `OnCompactionCompleted`. We assume blob file data would not be discarded during flush, which is not true. Consider the case:

0. There's a key "foo" in blob file b1.
1. GC rewrite b1 into b2 and insert "foo" to memtable.
2. Before memtable is flushed, user delete "foo".
3. After flush, "foo" is removed from the output of flush, since it's deleted. The fact that "foo" is discarded is not reflected in b2's discardable size.

**Comment by yiwu-arbug** [comment]
I don't know how to fix this one, since `OnFlushCompleted` don't know what's the input data size for each blob file.

**Comment by wujy-cs** [comment]
I think we can use output data size to get discardable_size of flushed blob file:

0. If discardable_size == 0, then discardable_size = file_size - output_size
1. if discardable_size > 0, then discardable_size = discardable_size - output_size ( for those blob file indexes that not be flushed completely in one run )

**Comment by yiwu-arbug** [comment]
@wujy-cs sounds reasonable. So basically, a) mark the latest memtable id (say, M) after GC finish, b) wait till M flushed, then subtract file size by sum of output size of all memtables?

**Comment by wujy-cs** [comment]
@yiwu-arbug 
Yes,. And we can do it in OnFlushCompleted without distinguishing gc rewritten blob file and flushed blob file, since outpu_size of flushed blob file always equals to file_size.
